### DISCUSSION TITLE
Why does it need admin priviledges?

### DISCUSSION THREAD

**Discussion by dc46and2** [iXab3r/MicSwitch#74] [General]
Thanks for the very useful program. On Windows 10, I get a UAC prompt every time I start MicSwitch. I'm curious: what is MicSwitch doing that requires admin privileges? Would it be possible to run as a normal user with some features disabled?

**Comment by iXab3r** [comment]
Hi ! I'll look into it, although it may be that some system functions related to getting device list or muting/unmuting a device will not be available without admin privileges, thus making software mostly unusable

**Reply by VortiFred** [reply]
Hello,
I force RunAsInvoker policy before MicSwitch starts.
Set the __COMPAT_LAYER=RUNASINVOKER environment variable to avoid this UAC behavior.
Elevation privileges is not necessary to use MicSwitch. I think this behavior is related to the update feature.
